modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/seedlist/HistoricalSnapshotOrdinalEntries.scala

```diff
@@ -0,0 +1,76 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.MonadThrow
+import cats.syntax.all._
+import org.tessellation.schema.SnapshotOrdinal
+import derevo.cats.eqv
+import derevo.derive
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.NonNegative
+import eu.timepit.refined.refineV
+import fs2.data.csv.{CellDecoder, DecoderError}
+
+import scala.util.control.NoStackTrace
+
+@derive(eqv)
+case class HistoricalSnapshotOrdinalEntries(values: List[HistoricalSnapshotOrdinalEntry])
+
+case object InvalidRange extends NoStackTrace
+
+object HistoricalSnapshotOrdinalEntries {
+
+  implicit val historicalSnapshotOrdinalEntriesCellDecoder: CellDecoder[Option[HistoricalSnapshotOrdinalEntries]] =
+    CellDecoder.stringDecoder.emap { s =>
+      mkSnapshotOrdinalTimeline(s).attempt
+        .map(
+          _.leftMap(_ => new DecoderError("Unable to parse the ordinal timeline"))
+            .map(_.some)
+        )
+        .get
+    }
+
+  private val valueDelimiter = "-"
+  private val entryDelimiter = ";"
+
+  def toRange[G[_]: MonadThrow](str: String): G[(Long, Long)] =
```

Comment:
why `G[_]`? We use `F[_]` everywhere and `G` if it's a second type which is not the case here.

Reply:
Reworked to

```scala
  def toRangeBoundary(str: String): Either[Throwable, RangeBoundary] = {
    val values = str
      .split(valueDelimiter)
      .map(_.trim)
      .map(
        _.toLongOption
          .flatMap(refineV[NonNegative](_).toOption)
      )

    values match {
      case Array(Some(start), Some(end), _*) =>
        Either.cond(
          start <= end,
          RangeBoundary(start = start, end = end),
          InvalidRange
        )
      case Array(Some(start), _*) =>
        RangeBoundary(
          start = start,
          end = Refined.unsafeApply[Long, NonNegative](Long.MaxValue)
        ).asRight
      case _ => InvalidRange.asLeft
    }
  }
```